mayastor-test/e2e/uninstall/uninstall_test.go

```diff
@@ -0,0 +1,193 @@
+package basic_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"path"
+	"runtime"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	coreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/deprecated/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var cfg *rest.Config
+var k8sClient client.Client
+var k8sManager ctrl.Manager
+var testEnv *envtest.Environment
+
+/// Examine the nodes in the k8s cluster and return
+/// the IP address of the master node (if one exists),
+/// The assumption is that the test-registry is accessible via the IP addr of the master,
+/// or any node in the cluster if the master noe does not exist
+/// TODO Refine how we workout the address of the test-registry
+func getRegistryAddress() (string, error) {
+	var master = ""
+	nodeList := coreV1.NodeList{}
+	if (k8sClient.List(context.TODO(), &nodeList, &client.ListOptions{}) != nil) {
+		return master, errors.New("failed to list nodes")
+	}
+	nodeIPs := make([]string, len(nodeList.Items))
+	for ix, k8node := range nodeList.Items {
+		for _, k8Addr := range k8node.Status.Addresses {
+			if k8Addr.Type == coreV1.NodeInternalIP {
+				nodeIPs[ix] = k8Addr.Address
+				for label := range k8node.Labels {
+					if label == "node-role.kubernetes.io/master" {
+						master = k8Addr.Address
+					}
+				}
+			}
+		}
+	}
+
+	/// TODO Refine how we workout the address of the test-registry
+
+	/// If there is master node, use its IP address as the registry IP address
+	if len(master) != 0 {
+		return master, nil
+	}
+
+	if len(nodeIPs) == 0 {
+		return "", errors.New("no usable nodes found")
+	}
+
+	/// Choose the IP address of first node in the list as the registry IP address
+	return nodeIPs[0], nil
+}
+
+// Encapsulate the logic to find where the deploy yamls are
+func getDeployYamlDir() string {
+	_, filename, _, _ := runtime.Caller(0)
+	return path.Clean(filename + "/../../../../deploy")
+}
+
+// Helper for passing yaml from the deploy directory to kubectl
+func deleteDeployYaml(filename string) {
```

Comment:
I've just written a function exactly like this. Maybe we need we should gather commonly-used functions together.

Reply:
Sure I will do that as another PR, as there is stuff that needs to be shared from the install test as well.